### PR TITLE
Implement QIteratedSimplex.

### PR DESCRIPTION
**Pull request by drwells**
A limited version of QIterated for simplex meshes.

There is a way to implement this for non-powers of 2 with triangles - see

https://github.com/dealii/dealii/blob/5b886bc914d6c5042492b0decbe31c78b2515268/include/deal.II/numerics/data_out_dof_data.templates.h#L107-L201

but for now I am mostly interested in using it with tets so I didn't bother adding it yet.

### CODE REVIEW

**Review by peterrum — APPROVED** [review]
Looks good except of the implicit casts!

**Comment by drwells** [comment]
Fixed!

**Review by peterrum — APPROVED** [review]

File: source/base/quadrature_lib.cc [hunk 1]
```diff
@@ -1739,6 +1746,82 @@ QWitherdenVincentSimplex<dim>::QWitherdenVincentSimplex(
...
+          setup_qiterated_1D(base_quad, n_copies);
+        break;
+      case 2:
+      case 3:
+        {
+          const auto n_refinements =
+            static_cast<unsigned int>(std::log2(n_copies));
+          Assert(std::abs(std::pow(2.0, n_refinements) - n_copies) <
+                   n_copies * 1e-10,
+                 ExcMessage("The number of copies must be a power of 2."));
```
Comment (masterleinad) [inline source/base/quadrature_lib.cc:1788]: Does the rounding always work correctly here or should we add `.5`?
```suggestion
          const auto n_refinements =
            static_cast<unsigned int>(std::log2(n_copies) + .5);
          Assert(std::abs(std::pow(2.0, n_refinements) - n_copies) <
                   n_copies * 1e-10,
                 ExcMessage("The number of copies must be a power of 2."));
```

Comment (drwells) [reply source/base/quadrature_lib.cc:1788]: Good point - let me throw in a `std::round` to make sure this does what I think it does.

**Review by masterleinad — COMMENTED** [review]
Apart from the log2 question, this looks good to me.

**Review by masterleinad — APPROVED** [review]
Looks good to me!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**